test/e2e/egress_traffic_test.go

```diff
@@ -56,7 +57,20 @@ func TestEgressTraffic(t *testing.T) {
 		t.Fatalf("Can't get internal request domain: service.Route.Status.URL is nil")
 	}
 	t.Log(service.Route.Status.URL.Host)
-	response, err := sendRequest(t, clients, test.ServingFlags.ResolvableDomain, service.Route.Status.URL.Host)
+
+	domain := service.Route.Status.URL.Host
+	if _, err = pkgTest.WaitForEndpointState(
+		clients.KubeClient,
+		t.Logf,
+		domain,
+		v1a1test.RetryingRouteInconsistency(pkgTest.IsStatusOK),
+		"HTTPProxy",
+		test.ServingFlags.ResolvableDomain); err != nil {
+		t.Fatalf("Failed to start endpoint of httpproxy: %v", err)
+	}
+	t.Log("httpproxy is ready.")
+
+	response, err := sendRequest(t, clients, test.ServingFlags.ResolvableDomain, domain)
```

Comment:
This is kinda redundant now, right? `WaitForEndpointState` does effectively the same thing.

Reply:
Yes, agree. Removed the redundant part.